uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -530,26 +599,197 @@ define([
+       */
+      function initDates() {
+        if (canEdit()) {
+          vm.uiOptions.fromDate = convertDateFormatFromServer(vm.leaveRequest.from_date);
+          vm.uiOptions.toDate = convertDateFormatFromServer(vm.leaveRequest.to_date);
+
+          vm.onDateChange(vm.uiOptions.fromDate, 'from');
+          vm.onDateChange(vm.uiOptions.toDate, 'to');
+        }
+      }
+
+      /**
+       * Initialize status
+       */
+      function initStatus() {
+        if (canEdit()) {
+          //set it before vm.leaveRequestStatuses gets filtered
+          vm.statusLabel = vm.leaveRequestStatuses[vm.leaveRequest.status_id].label;
+          if (vm.role === 'manager') {
+            setStatuses();
+          }
+        } else {
+          vm.leaveRequest.status_id = valueOfRequestStatus('waiting_approval');
+        }
+      }
+
+      /**
+       * Initialize contact
+       *
+       * {Promise}
+       */
+      function initContact() {
+        if (vm.role === 'manager') {
+          return Contact.find(vm.leaveRequest.contact_id)
+            .then(function (contact) {
+              vm.contact = contact;
+            });
+        } else {
+          return $q.resolve({});
+        }
+      }
+
+      /**
+       * Sets leave requestion statuses
+       */
+      function setStatuses() {
+        var allowedStatuses = ['approved', 'more_information_requested', 'cancelled'];
+
+        if (vm.role === 'manager') {
+          //remove current status of leaverequest
+          _.remove(allowedStatuses, function (status) {
+            return status == vm.leaveRequestStatuses[vm.leaveRequest.status_id].name;
+          });
+
+          //filter vm.leaveRequestStatuses to contain statues relevant for manager to act
+          for (var key in vm.leaveRequestStatuses) {
+            var status = vm.leaveRequestStatuses[key];
+
+            if (!_.includes(allowedStatuses, status.name)) {
+              delete vm.leaveRequestStatuses[key];
+            }
+          }
+        }
+      }
+
+      /**
+       * Creates leaverequest
+       */
+      function createRequest() {
+        vm.leaveRequest.isValid()
+          .then(function () {
+            vm.leaveRequest.create()
+              .then(function () {
+                // refresh the list
+                postSubmit('LeaveRequest::new');
+              })
+              .catch(handleError);
+          })
+          .catch(handleError);
+      }
+
+      /**
+       * Updates the leaverequest
+       */
+      function updateRequest() {
+        vm.leaveRequest.update()
+          .then(function () {
+            postSubmit('LeaveRequest::updatedByManager');
+          })
+          .catch(handleError);
+      }
+
+      /**
+       * Called after successful submission of leave request
+       *
+       * @param {String} eventName name of the event to emit
+       */
+      function postSubmit(eventName) {
+        $rootScope.$emit(eventName, vm.leaveRequest);
+        vm.error = undefined;
```

Comment:
Setting `undefined` value to a variable does not look good, we should use `null`. `undefined` is when the variable is not initialized. So if we want to assign a value that means we are defining it. So we should use `null`.